proxy/envoy/config.go

```diff
@@ -184,13 +184,28 @@ func buildOutboundFilters(instances []*model.ServiceInstance, services []*model.
 	suffix := sharedInstanceHost(instances)
 	httpConfigs := make(HTTPRouteConfigs)
 
+	// get all the route rules applicable to the instances
+	rules := config.RouteRulesBySource("", instances)
+
```

Comment:
If instances is the set of all destination services in k8s, then this function basically does that rulesByDestination did.